src/Data/Singletons/Prelude/Semigroup.hs

```diff
@@ -227,9 +206,7 @@ $(singletonsOnly [d|
     enumFromTo (First a) (First b) = First `map` enumFromTo a b
     enumFromThenTo (First a) (First b) (First c) = First `map` enumFromThenTo a b c
 
-  -- deriving instance Functor First
-  instance Functor First where
-    fmap f (First x) = First (f x)
+  deriving instance Functor First
```

Comment:
This isn't the first time I've seen this declaration...

Reply:
Note that the [`First` in `Data.Semigroup`](https://hackage.haskell.org/package/base-4.11.1.0/docs/Data-Semigroup.html#t:First) is not the same [`First` as in `Data.Monoid`](https://hackage.haskell.org/package/base-4.11.1.0/docs/Data-Monoid.html#t:First).